### DISCUSSION TITLE
Is there any workaround for pointers in the Fact struct?

### DISCUSSION THREAD

**Discussion by mledezma-r7** [hyperjumptech/grule-rule-engine#263] [Q&A]
Is there any workaround (or plan to support) for pointers to primitive types in the Fact struct

Something like this

```
type Fact struct {
	Foo *string
	Bar *bool
}
```

The nature of my data requires value distinction after unmarshalling the following two sample json docs
1) '{"Foo": "", "Bar": false}'
2) '{}'

```
With a Fact struct with no pointers
type Fact struct {
	Foo string
	Bar bool
}
```
the resulting struct would be equivalent to this for both case 1 and 2
Fact{Foo: "", Bar: false}

If we go with the struct with pointer we would get the following equivalents
1) 
```
Foo := ""
Bar := false
Fact{Foo: &Foo, Bar: &Bar}
```

2)
`Fact{Foo: nil, Bar: nil}`

Problem I'm facing is that I'm not able to pull (and evaluate) the values pointed by the pointers

Keep in mind I'm starting with go, so it might be a silly question
Thank you

**Comment by newm4n** [comment]
As of now, we have not planned to implement evaluation logic for pointer for facts  other than struct.
Its a good idea though. Could you create an issue ticket for this ?
